vsock/src/thread_backend.rs

```diff
@@ -0,0 +1,307 @@
+// SPDX-License-Identifier: Apache-2.0 or BSD-3-Clause
+
+#![deny(missing_docs)]
+
+use super::{
+    rxops::*,
+    vhu_vsock::{
+        ConnMapKey, Error, Result, VSOCK_HOST_CID, VSOCK_OP_REQUEST, VSOCK_OP_RST,
+        VSOCK_TYPE_STREAM,
+    },
+    vhu_vsock_thread::VhostUserVsockThread,
+    vsock_conn::*,
+};
+use log::{info, warn};
+use std::{
+    collections::{HashMap, HashSet, VecDeque},
+    os::unix::{
+        net::UnixStream,
+        prelude::{AsRawFd, FromRawFd, RawFd},
+    },
+};
+use virtio_vsock::packet::VsockPacket;
+use vm_memory::bitmap::BitmapSlice;
+
+// TODO: convert UnixStream to Arc<Mutex<UnixStream>>
```

Comment:
Why not convert now? Can we have multiple threads now?

Reply:
Nope, we have a single thread and I think that was an old TODO since `VsockThreadBackend` is supposed to be per-thread (IIUC).
Anyway, if we will move to multi-thread, I think we need to handle also other maps, so I would remove that one if you and  @harshanavkis agree.

Reply:
fine by me